scripts/scripts.js

```diff
@@ -66,6 +68,61 @@ function buildAutoBlocks(main) {
   }
 }
 
+/**
+ * Decorates all sections in a container element.
+ * @param {Element} main The container element
+ */
+function decorateSections(main) {
+  main.querySelectorAll(':scope > div').forEach((section) => {
+    const wrappers = [];
+    let defaultContent = false;
+
+    [...section.children].forEach((e) => {
+      if (e.classList.contains('richtext')) {
+        e.className = 'default-content-wrapper';
+        wrappers.push(e);
+      } else if (e.getAttribute('data-aue-type') === 'media') {
+        const wrapper = document.createElement('div');
+        wrapper.classList.add('default-content-wrapper');
+        wrappers.push(wrapper);
+        wrapper.append(e);
+      } else if (e.tagName === 'DIV' || !defaultContent) {
+        const wrapper = document.createElement('div');
+        wrappers.push(wrapper);
+        defaultContent = e.tagName !== 'DIV';
+        if (defaultContent) wrapper.classList.add('default-content-wrapper');
+        wrapper.append(e);
+      } else {
+        wrappers[wrappers.length - 1].append(e);
+      }
+    });
```

Comment:
I think this is fine - the original in `aem.js` is also not super easy to read without knowing what it does. Maybe we could simplify it somehow by splitting into several passes / using query selectors, but I prefer leaving it like this since compared to the original `aem.js` it looks recognizable + some additional ifs.

Reply:
@hannessolo I simplified the logic a little, it is now also closer to the behaviour on aem.page/aem.live